src/KingSystem/World/worldShootingStarMgr.cpp

```diff
@@ -1 +1,164 @@
 #include "KingSystem/World/worldShootingStarMgr.h"
+#include <container/seadObjArray.h>
+#include <random/seadGlobalRandom.h>
+#include "KingSystem/ActorSystem/actActorCreator.h"
+#include "KingSystem/ActorSystem/actBaseProcHeapMgr.h"
+#include "KingSystem/ActorSystem/actInstParamPack.h"
+#include "KingSystem/GameData/gdtManager.h"
+#include "KingSystem/World/worldManager.h"
+
+namespace ksys::world {
+
+static bool sScheduled = true;
+
+static struct {
+    int end = 3;
+    int start = 21;
+    int unk = 4;
+} sHours;
+
+static sead::Vector3f sStarProperty{};
+
+ShootingStarMgr::ShootingStarMgr() = default;
+
+ShootingStarMgr::~ShootingStarMgr() = default;
+
+void ShootingStarMgr::init_(sead::Heap* heap) {
+    if (Manager::instance()->isDemo())
+        sScheduled = false;
+    else
+        initSchedule();
+}
+
+// NON_MATCHING
+void ShootingStarMgr::initSchedule() {
+    if (sHours.start >= sHours.end) {
+        sead::FixedObjArray<u32, 24> validHours;
+
+        if (sHours.start <= 23) {
+            for (int i = 0; i < 24; ++i) {
+                validHours.emplaceBack(sHours.start);
+            }
+        }
+        if (sHours.end > 1) {
+            for (int i = 0; i < 23; ++i) {
+                validHours.emplaceBack(i);
+            }
+        }
+        auto rand = sead::Random();
+        validHours.shuffle(&rand);
+        mFallHour = *validHours[0];
+    } else {
+        mFallHour = sead::GlobalRandom::instance()->getS32Range(sHours.start, sHours.end - 1);
+    }
+    mFallMinute = sead::GlobalRandom::instance()->getU32(59);
+    mInitialised = true;
+}
+
+void ShootingStarMgr::calc_() {
+    if (Manager::instance() != nullptr) {
+        if (!mIsMainField)
+            return;
+        if (sScheduled) {
+            if (Manager::instance()->isDemo()) {
+                sScheduled = false;
+                return;
+            }
+            TimeMgr* tm = Manager::instance()->getTimeMgr();
+            if (tm != nullptr) {
+                int hour = tm->getHour();
+                if (mInitialised) {
+                    if (hour == mFallHour && mFallMinute < tm->getMinute())
+                        mNeedSpawnStar = true;
+                } else if (Manager::instance() != nullptr) {
+                    if (isScheduledTime())
+                        initSchedule();
+                }
+            }
+        }
+    }
+    if (mNeedSpawnStar && checkCamera()) {
+        spawnShootingStar();
+        mInitialised = false;
+        mNeedSpawnStar = false;
+    }
+}
+
+bool ShootingStarMgr::isScheduledTime() {
+    bool result = false;
+    if (Manager::instance() != nullptr) {
+        TimeMgr* tm = Manager::instance()->getTimeMgr();
```

Comment:
```suggestion
        auto* tm = Manager::instance()->getTimeMgr();
```
(I think the function and variable names make it pretty clear that this is indeed a TimeMgr)